gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StateStoreBasedWatermarkStorage.java

```diff
@@ -50,7 +50,12 @@
   public static final String WATERMARK_STORAGE_TYPE_KEY ="streaming.watermarkStateStore.type";
   public static final String WATERMARK_STORAGE_TYPE_DEFAULT ="zk";
   public static final String WATERMARK_STORAGE_CONFIG_PREFIX="streaming.watermarkStateStore.config.";
-  private static final String WATERMARK_STORAGE_PREFIX="streamingWatermarks:";
+
+  /**
+   * The prefix need to be fitting with different storage layer. e.g. some of characters won't be allowed in URI and
```

Comment:
A bit of word-smithing: "A watermark prefix that is compatible with different watermark storage implementations. As such, this prefix should not include any characters disallowed in a {@link URI}."

Reply:
Thanks, addressed.